src/mscorlib/shared/System/IO/Win32Marshal.cs

```diff
@@ -84,7 +90,7 @@ internal static Exception GetExceptionForWin32Error(int errorCode, string path)
                     return new OperationCanceledException();
 
                 default:
-                    return new IOException(GetMessage(errorCode), MakeHRFromErrorCode(errorCode));
+                    return new IOException(Interop.Kernel32.GetMessage(errorCode), MakeHRFromErrorCode(errorCode));
```

Comment:
If you exercise this codepath by calling `System.IO.Win32Marshal.GetExceptionForWin32Error(1234567);`, it will fail with:

```
jkotas-mac:OSX.x64.Debug jkotas$ ./corerun test.exe 

Unhandled Exception: System.DllNotFoundException: Unable to load DLL 'kernel32.dll': The specified module or one of its dependencies could not be found.
 (Exception from HRESULT: 0x8007007E)
   at Interop.Kernel32.FormatMessage(Int32 dwFlags, IntPtr lpSource, UInt32 dwMessageId, Int32 dwLanguageId, StringBuilder lpBuffer, Int32 nSize, IntPtr[] arguments)
   at Interop.Kernel32.TryGetErrorMessage(IntPtr moduleHandle, Int32 errorCode, StringBuilder sb, String& errorMsg)
   at Interop.Kernel32.GetMessage(IntPtr moduleHandle, Int32 errorCode)
   at Interop.Kernel32.GetMessage(Int32 errorCode)
   at System.IO.Win32Marshal.GetExceptionForWin32Error(Int32 errorCode, String path)
   at System.IO.Win32Marshal.GetExceptionForWin32Error(Int32 errorCode)
```

Apparently, we do not have any coverage in our tests for this codepath on Unix. But there may be some rare situations where it will end up being called. I would be useful to not break it.

The options I can think of:
1. Dumb it down on Unix so that it does not call Kernel32. Make it a message with error code or something.
2. Or fix the ifdefs so that the shared code still reroutes the Kernel32 calls into the CoreCLR PAL.

Either one is fine with me.